src/java/org/apache/cassandra/index/Index.java

```diff
@@ -593,10 +602,296 @@ default boolean supportsReplicaFilteringProtection(RowFilter rowFilter)
      */
     public interface Searcher
     {
+        /**
+         * Returns the {@link ReadCommand} for which this searcher has been created.
+         *
+         * @return the base read command
+         */
+        ReadCommand command();
```

Comment:
This breaks API compatibility without a default implementation, which is fine if this is going to ship in a major.